src/Microsoft.AspNet.Server.Kestrel/Filter/SocketInputStream.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Server.Kestrel.Http;
+
+namespace Microsoft.AspNet.Server.Kestrel.Filter
+{
+    public class SocketInputStream : Stream
+    {
+        private static Task _empty = Task.FromResult<object>(null);
+
+        private readonly SocketInput _socketInput;
+
+        public SocketInputStream(SocketInput socketInput)
+        {
+            _socketInput = socketInput;
+        }
+
+        public override bool CanRead => false;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => true;
```

Comment:
This is inverted and worth a comment. The fact that we're using Write to Write to the input stream isn't exactly intuitive.